server/src/main/java/org/apache/iotdb/db/newsync/receiver/collector/Collector.java

```diff
@@ -19,4 +19,189 @@
  */
 package org.apache.iotdb.db.newsync.receiver.collector;
 
-public class Collector {}
+import org.apache.iotdb.db.concurrent.IoTDBThreadPoolFactory;
+import org.apache.iotdb.db.concurrent.ThreadName;
+import org.apache.iotdb.db.exception.metadata.IllegalPathException;
+import org.apache.iotdb.db.newsync.pipedata.PipeData;
+import org.apache.iotdb.db.newsync.utils.BufferedPipeDataBlockingQueue;
+import org.apache.iotdb.db.newsync.utils.SyncConstant;
+import org.apache.iotdb.db.newsync.utils.SyncPathUtil;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+/** scan sync receiver folder and load pipeData into IoTDB */
+public class Collector {
+
+  private static final Logger logger = LoggerFactory.getLogger(Collector.class);
+  private static final int WAIT_TIMEOUT = 2000;
+  private ExecutorService executorService;
+  private Map<String, Future> taskFutures;
+
+  public Collector() {
+    taskFutures = new HashMap<>();
```

Comment:
HashMap is not thread-safe. use concurrentHashMap plz.

Reply:
fixed